docs/features/compilation-from-portable-pdb.md

```diff
@@ -0,0 +1,341 @@
+## compilation-from-portable-pdb
+
+Compilation from portable PDBs today is not completely possible, but is desirable in order to help reconstruct a compilation from source provided via source link or embedded in a pdb. Motivation is derived from [roslyn 41395](https://github.com/dotnet/roslyn/issues/41395).
+
+This document is restricted to the following assumptions:
+
+1. The benefit is for builds with `-deterministic` and published to the symbol server.
+2. Source generator and analyzer references are not needed for this task. They may be useful, but are out of scope for this document.
+3. Any storage capacity used for PDBs and source should not impact this feature, such as compression algorithm.
+4. Only Portable PDB files will be included for this spec. We can expand this feature past these once it is implemented and proven needed elsewhere.
+
+This document will provide the expanded specification to the Portable PDB format. Any additions to that format will be ported to expand documentation provided in [dotnet-runtime](https://github.com/jnm2/dotnet-runtime/blob/26efe3467741fe2a85780b2d2cd18875af6ebd98/docs/design/specs/PortablePdb-Metadata.md#source-link-c-and-vb-compilers).
+
+## List of Compiler Flags
+
+#### CSharp Flags That Can Be Derived From PDB or Assembly
+
+* baseaddress
+* checksumalgorithm
+* debug
+* deterministic
+* embed
+* filealign
+* highentropyva
+* link
+    - Represented by a metadata reference with `EmbededInteropTypes=true` 
+* linkresource
+    - Will be represented in metadata reference embedded in pdb
+* main
+    - Already stored in PDB as the entry point token
+* moduleassemblyname
+* modulename
+* nowin32manifest
+* pdb
+* platform
+* publicsign
+* resource
+    - Will be represented in metadata reference embedded in pdb
+* subsystemversion
+* target
+* win32icon
+* win32manifest
+* win32res
+
+#### CSharp Flags Not Included
+
+* appconfig
```

Comment:
This impacts assembly identity resolution. If this is omitted then it's possible there won't be enough info in the PDB to successfully re-creeate compilations.

Reply:
Is the appconfig file included in embedded resources or sourcelink somewhere? If we include this, it may not be possible to get the same file again

Reply:
No it's not included. It's simply consumed at compilation time. But it is necessary to reproduce the compilation given the same inputs (assuming of course the appconfig was provided because it was needed to resolve reference dependency issues)

Reply:
Yes, you're right. I forgot about this one. It's needed in order to create `AssemblyIdentityComparer`. The only data needed from the config file are stored in [`AssemblyPortabilityPolicy`](http://sourceroslyn.io/#Microsoft.CodeAnalysis/Desktop/AssemblyPortabilityPolicy.cs,14). There are two bools we need to serialize. We can, for example, serialize this as option `portability-policy` = number in [0..3] (1 bit for each bool).